simple-backend/nlpviewer_backend/lib/stave_viewer.py

```diff
@@ -0,0 +1,315 @@
+worry about username, password, credentials, and permissions, so that it is
+simple and straightforward for users to boot up Stave for easy-visualization.
+
+Package Requirements:
+    tornado == 6.1
+    django == 3.2
+
+Environment:
+    PYTHONPATH: Absolute path (or relative path from PYTHONPATH)
+        to django backend folder should be inserted into PYTHONPATH.
+        Example: "stave/simple-backend/". Deprecated in future update.
+"""
+
+import os
+import sys
+import json
+import errno
+import logging
+import asyncio
+import threading
+import webbrowser
+from typing import Dict, Set, Any, List
+import requests
+
+import django
+from django.core.wsgi import get_wsgi_application
+
+from tornado.web import FallbackHandler, StaticFileHandler, \
+    RequestHandler, url, Application
+from tornado.httpserver import HTTPServer
+from tornado.ioloop import IOLoop
+from tornado.wsgi import WSGIContainer
+
+from .stave_project import StaveProjectReader
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "StaveViewer"
+]
+
+
+class StaveViewer:
+    """
+    StaveViewer allows users to start and control a Stave instance in viewer
+    mode. Example usage:
+
+        sv = StaveViewer(
+            build_path = os.path.join(stave_path, "build/")
+            project_path = project_path
+        )
+        sv.run()
+        sv.open()
+
+    The codes above initialize a StaveViewer instance by passing the Stave
+    build path(deprecated in future update) and project path. By calling
+    `run()`, StaveViewer starts a tornado server on a newly created thread.
+    `open()` method will open the browser with the visualization.
+    
+    Note: Currently the viewer mode is achieved by rerouting the django
+    backend requests and it can be hard to maintain when structure changes
+    in Stave. Refactoring is welcome to improve code logic and maintanence.
+    """
+
+    def __init__(self,
+        build_path: str,
+        project_path: str = '',
+        host: str = "localhost",
+        port: int = 8888,
+        thread_daemon: bool = False,
+        in_viewer_mode: bool = True
+    ):
+        """
+        Initialize StaveViewer with input paramaters.
+
+        Args:
+            build_path: Absolute path (or relative path from PYTHONPATH)
+                to stave build folder. Example: "stave/build/".
+            project_path: Path to the project directory for rendering.
+                Default to the current path ''.
+            host: host name for Stave server. Default value is `localhost`.
+            port: port number for Stave server. Default value is 8888.
+            thread_daemon: Set whether the thread is daemonic.
+                Default to False.
+            in_viewer_mode: Enable viewer mode of Stave. If False, StaveViewer
+                will start a standard Stave instance. Default to True.
+        """
+        self._project_reader: "StaveProjectReader"
```

Comment:
Why adding " on the class? `StaveProjectReader` is in another module and should be available here.